src/main/java/org/openstreetmap/atlas/checks/validation/tag/HighwayMissingNameAndRefTagCheck.java

```diff
@@ -0,0 +1,376 @@
+    private static final String MISSING_BOTH_NAME_AND_REF_TAG_INSTRUCTIONS = "Way {0, number, #} is missing both name and ref tag. Way must contain either one.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections
+            .singletonList(MISSING_BOTH_NAME_AND_REF_TAG_INSTRUCTIONS);
+    private final double minAngleForNonContiguousWays;
+    private final HighwayTag minHighwayTag;
+    private static final String MIN_HIGHWAY_TAG_DEFAULT = "tertiary";
+    private static final double MIN_ANGLE_NON_CONTIGUOUS_WAYS = 30;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public HighwayMissingNameAndRefTagCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.minAngleForNonContiguousWays = configurationValue(configuration,
+                "min.nonContiguous.angle", MIN_ANGLE_NON_CONTIGUOUS_WAYS);
+        this.minHighwayTag = Enum.valueOf(HighwayTag.class,
+                this.configurationValue(configuration, "min.highway.type", MIN_HIGHWAY_TAG_DEFAULT)
+                        .toUpperCase());
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && ((Edge) object).isMainEdge()
+                && !isFlagged(object.getOsmIdentifier())
+                && ((Edge) object).highwayTag().isMoreImportantThanOrEqualTo(this.minHighwayTag)
+                && !HighwayTag.isLinkHighway(((Edge) object).highwayTag())
+                && !JunctionTag.isRoundabout(object) && !JunctionTag.isCircular(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        markAsFlagged(object.getOsmIdentifier());
+        final Map<String, String> tags = object.getTags();
+        if (this.highwayMissingBothNameAndRefTag(tags)
+                && !this.isConnectorWayToIgnore(((Edge) object).getMainEdge()))
+        {
+            return Optional.of(
+                    createFlag(object, this.getLocalizedInstruction(0, object.getOsmIdentifier())));
```

Comment:
We need to flag all Edges with the same OSM id together. Otherwise it will be nondeterministic which Edge from a Way gets flagged.

Reply:
addressed